compiler/libpc300/sc3.c

```diff
@@ -1209,6 +1236,8 @@ static int hier2(value *lval)
       rvalue(lval);
     invert();                   /* bitwise NOT */
     lval->constval=~lval->constval;
+    if (lval->ident==iVARIABLE || lval->ident==iARRAYCELL)
+      lval->ident=iEXPRESSION;
```

Comment:
These are long-standing bugs - but are they related to this change? If not, it's best to separate them for purposes of bisection.

Reply:
Actually, it looks like there are many individual bug changes in this patch...

Reply:
Should i just separate them? Or it's possible to change pr title instead? (Coz i just took arkshine style, he used 1 pr to update/fix few things in compiler).

Reply:
one more, 

> psychonic >
> https://builds.alliedmods.net/amxx/builders/win32-1.10

Reply:
@dvander need answers